src/test/java/unit/GcsBucketReferenced.java

```diff
@@ -0,0 +1,239 @@
+package unit;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static unit.GcsBucketControlled.listBucketResourceWithName;
+import static unit.GcsBucketControlled.listOneBucketResourceWithName;
+
+import bio.terra.cli.serialization.userfacing.resources.UFGcsBucket;
+import bio.terra.workspace.model.CloningInstructionsEnum;
+import com.google.cloud.storage.Bucket;
+import harness.TestCommand;
+import harness.baseclasses.SingleWorkspaceUnit;
+import harness.utils.ExternalGCSBuckets;
+import java.io.IOException;
+import java.util.List;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+/** Tests for the `terra resources` commands that handle referenced GCS buckets. */
+@Tag("unit")
+public class GcsBucketReferenced extends SingleWorkspaceUnit {
+
+  // external bucket to use for creating GCS bucket references in the workspace
+  private Bucket externalBucket;
+
+  @BeforeEach
+  @Override
+  protected void setupEachTime() throws IOException {
+    super.setupEachTime();
+    externalBucket = ExternalGCSBuckets.createBucket();
```

Comment:
do we need a new bucket for every test case or can we share one across all?

Reply:
good question. I'm not sure whether WSM checks for duplicate references or not. will check

Reply:
looks like it doesn't check, so we can use the same bucket over and over as long as the resource name is different.